hal/src/nRF52840/usart_hal.cpp

```diff
@@ -618,7 +618,7 @@ const auto UARTE1_INTERRUPT_PRIORITY = APP_IRQ_PRIORITY_HIGHEST;
 Usart* getInstance(HAL_USART_Serial serial) {
     static Usart usartMap[] = {
         {NRF_UARTE0, uarte0InterruptHandler, UARTE0_INTERRUPT_PRIORITY, NRF_TIMER3, NRF_PPI_CHANNEL4, TX, RX, CTS, RTS},
-        {NRF_UARTE1, uarte1InterruptHandler, UARTE1_INTERRUPT_PRIORITY, NRF_TIMER4, NRF_PPI_CHANNEL5, TX1, RX1, CTS1, RTS1}
+        {NRF_UARTE1, uarte1InterruptHandler, UARTE1_INTERRUPT_PRIORITY, NRF_TIMER1, NRF_PPI_CHANNEL5, TX1, RX1, CTS1, RTS1}
```

Comment:
We've run out of timer resource, I'd like to release the timer in button hal, we have below choice for button debounce timer, or you can add  other suggestion:
 - Use FreeRTOS timer in Device OS and use Timer 2 in bootloader
 - Use a high-precision timer to make a general software timer just like `app timer`, then we can use it in Bootloader and Device OS
 - Add a software timer handler in `HAL_SysTick_Handler()`, if we start the button timer, we set a timer flag, then we check the flag in `HAL_SysTick_Handler()`

I'd prefer the third one, since the handler won't take a long time.

**Timer resource:**
```
RTC0: Softdevice
RTC1: FreeRTOS
RTC2: OpenThread

Timer0: Softdevice
Timer1: Radio
Timer2: Button_hal
Timer3: Usart
Timer4: Usart
```

Reply:
Agreed, I don't think we have any other choice at the moment. Moreover on Gen 3 devices this shouldn't cause any issues for FreeRTOS, since FreeRTOS is run off of RTC timer instead of SysTick.